src/components/v2/appDetails/sourceInfo/environmentStatus/AppStatusDetailModal.tsx

```diff
@@ -18,12 +20,44 @@ interface NodeStreamMap {
     version: string
 }
 
-function AppStatusDetailModal({ close, appStreamData, showAppStatusMessage }: { close: () => void; appStreamData: any; showAppStatusMessage?: boolean }) {
+const STATUS_SORTING_ORDER = {
+    [NodeStatus.Degraded]: 1,
+    [NodeStatus.Progressing]: 2,
+    [NodeStatus.Healthy]: 3,
+}
+
+function AppStatusDetailModal({
+    close,
+    appStreamData,
+    showAppStatusMessage,
+}: {
+    close: () => void
+    appStreamData: any
+    showAppStatusMessage?: boolean
+}) {
     const _appDetails = IndexStore.getAppDetails()
 
     const nodes: AggregatedNodes = useMemo(() => {
         return aggregateNodes(_appDetails.resourceTree?.nodes || [], _appDetails.resourceTree?.podMetadata || [])
     }, [_appDetails])
+    const nodesKeyArray = Object.keys(nodes?.nodes || {})
```

Comment:
can we change this to something like - `nodes?.nodes ? Object.keys(nodes.nodes) : []`?